File: flutter_baidu_yingyan_trace-1.0.0/android/src/main/java/com/baidu/flutter/trace/model/entity/UpdateEntityOption.java
package com.baidu.flutter.trace.model.entity;

import java.util.HashMap;
import java.util.Map;

import com.baidu.flutter.trace.model.BaseOption;
import com.baidu.trace.api.entity.UpdateEntityRequest;

/**
 * 更新Entity请求参数
 *
 * @author baidu
 */
public final class UpdateEntityOption extends BaseOption {

    /**
     * entity唯一标识
     */
    private String entityName;

    /**
     * entity的描述
     */
    private String entityDesc;

    /**
     * 自定义属性
     */
    private HashMap<String, String> customColumns;

    public String getEntityName() {
        return entityName;
    }

    /**
     * 设置entity名称
     *
     * @param entityName
     */
    public void setEntityName(String entityName) {
        this.entityName = entityName;
    }

    public String getEntityDesc() {
        return entityDesc;
    }

    /**
     * 设置entity的描述
     *
     * @param entityDesc
     */
    public void setEntityDesc(String entityDesc) {
        this.entityDesc = entityDesc;
    }

    public Map<String, String> getColumns() {
        return customColumns;
    }

    /**
     * 设置自定义属性
     *
     * @param columns
     */
    public void setColumns(HashMap<String, String> columns) {
        this.customColumns = columns;
    }

    public UpdateEntityOption() {
        super();
    }

    /**
     * @param tag       请求标识
     * @param serviceId 轨迹服务ID
     */
    public UpdateEntityOption(int tag, long serviceId) {
        super(tag, serviceId);
    }

    /**
     * @param tag        请求标识
     * @param serviceId  轨迹服务ID
     * @param entityName entity标识
     */
    public UpdateEntityOption(int tag, long serviceId, String entityName) {
        super(tag, serviceId);
        this.entityName = entityName;
    }

    /**
     * 更新Entity请求
     *
     * @param tag        请求标识
     * @param serviceId  轨迹服务ID
     * @param entityName entity标识
     * @param entityDesc entity描述
     * @param columns    自定义属性字段
     */
    public UpdateEntityOption(int tag, long serviceId, String entityName, String entityDesc,
                              HashMap<String, String> columns) {
        super(tag, serviceId);
        this.entityName = entityName;
        this.entityDesc = entityDesc;
        this.customColumns = columns;
    }

    public UpdateEntityRequest toUpdateEntityRequest() {
        UpdateEntityRequest updateEntityRequest = new UpdateEntityRequest();
        updateEntityRequest.setTag(tag);
        updateEntityRequest.setServiceId(serviceId);
        updateEntityRequest.setEntityName(entityName);
        updateEntityRequest.setEntityDesc(entityDesc);
        updateEntityRequest.setColumns(customColumns);
        return updateEntityRequest;
    }

    @Override
    public String toString() {
        return "UpdateEntityOption [tag=" + tag + ", serviceId=" + serviceId + ", entityName="
                + entityName
                + ", entityDesc=" + entityDesc + ", columns=" + customColumns + "]";
    }

}

File: flutter_baidu_yingyan_trace-1.0.0/lib/src/model/entity/update_entity_option.dart
import 'package:flutter_baidu_yingyan_trace/src/model/base_option.dart';

/// 更新Entity请求参数
class UpdateEntityOption extends BaseOption {

  ///  entity名称，作为终端实体的唯一标识，必填
  /// 同一service服务中entity_name不可重复。一旦创建，entity_name 不可更新。
  /// 命名规则：仅支持中文、英文大小字母、英文下划线"_"、英文横线"-"和数字。 entity_name 和 entity_desc 支持联合模糊检索。
  String ?entityName;

  /// entity的可读性描述，选填。
  /// 命名规则：仅支持中文、英文大小字母、英文下划线"_"、英文横线"-"和数字。entity_name 和 entity_desc 支持联合模糊检索
  String ?entityDesc;

  /// 开发者自定义字段，选填
  /// 字典中的key必须是已经通过鹰眼的轨迹管理台创建过的属性字段才有效
  Map<String, String> ?customColumns;

  UpdateEntityOption(
      {required int tag,
      required int serviceId,
      required this.entityName,
      this.entityDesc,
      this.customColumns})
      : assert(entityName != null),
        super(tag: tag, serviceId: serviceId);

  /// map => UpdateEntityOption
  UpdateEntityOption.fromMap(Map map)
      : assert(map != null,
  'Construct a UpdateEntityOption，The parameter map cannot be null !'),
        super.fromMap(map) {
    entityName = map['entityName'];
    entityDesc = map['entityDesc'];
    customColumns = map['customColumns'] == null ? null :
    map['customColumns'] as Map<String, String>;
  }

  /// UpdateEntityOption => map
  @override
  Map toMap() {
    Map map = super.toMap();
    map['entityName'] = this.entityName;
    map['entityDesc'] = this.entityDesc;
    map['customColumns'] = this.customColumns;
    return map;
  }

  @override
  String toString() {
    return 'UpdateEntityOption{tag: $tag, serviceId: $serviceId, '
        'entityName: $entityName, entityDesc: $entityDesc, '
        'customColumns: $customColumns}';
  }
}